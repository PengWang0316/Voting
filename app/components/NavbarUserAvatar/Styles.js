import { StyleSheet } from 'react-native';

export default StyleSheet.create({
  rootView: {
    marginRight: 15,
  },
});
