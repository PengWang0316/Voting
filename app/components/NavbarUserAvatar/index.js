import NavbarUserAvatar from './NavbarUserAvatar';

export default NavbarUserAvatar;
