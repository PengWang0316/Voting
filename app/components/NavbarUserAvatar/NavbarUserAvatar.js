import React from 'react';
import { View } from 'react-native';
import { connect } from 'react-redux';
import { Icon, Avatar } from 'react-native-elements';

import Styles from './Styles';

const NAVIGATION_LOGIN_NAME = 'Login';
const NAVIGATION_LOGIN_INFO_NAME = 'UserInfo';

type Props = {
  navigation: Object,
  user: Object,
};

export const NavbarUserAvatar = ({ navigation, user }: Props) => (
  <View style={Styles.rootView}>
    {!user._id && <Icon type="font-awesome" name="user-circle" color="white" onPress={() => navigation.navigate(NAVIGATION_LOGIN_NAME)} />}
    {user._id && <Avatar size={35} rounded icon={{ type: 'font-awesome', name: 'user-secret' }} onPress={() => navigation.navigate(NAVIGATION_LOGIN_INFO_NAME)} />}
  </View>
);
/* istanbul ignore next */
const mapStateToProps = state => ({
  user: state.user,
});
export default connect(mapStateToProps, null)(NavbarUserAvatar);
