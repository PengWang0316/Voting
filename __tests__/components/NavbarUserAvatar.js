import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import { NavbarUserAvatar } from '../../app/components/NavbarUserAvatar/NavbarUserAvatar';

jest.mock('react-native-elements', () => ({
  Avatar: () => 'Avatar',
  Icon: () => 'Icon',
}));

describe('NavbarUserAvatar', () => {
  const defaultProps = {
    navigation: {
      navigate: jest.fn(),
    },
    user: {},
  };
  const getShallowComponent = (props = defaultProps) => shallow(<NavbarUserAvatar {...props} />);

  test('Click Icon', () => {
    const component = getShallowComponent();
    component.find('Icon').simulate('press');
    expect(defaultProps.navigation.navigate).toHaveBeenCalledTimes(1);
    expect(defaultProps.navigation.navigate).toHaveBeenLastCalledWith('Login');
  });

  test('Click Avatar', () => {
    const component = getShallowComponent({ ...defaultProps, user: { _id: 'id' } });
    component.find('Avatar').simulate('press');
    expect(defaultProps.navigation.navigate).toHaveBeenCalledTimes(2);
    expect(defaultProps.navigation.navigate).toHaveBeenLastCalledWith('UserInfo');
  });

  test('Snapshot not user _id', () => expect(renderer.create(<NavbarUserAvatar {...defaultProps} />).toJSON()).toMatchSnapshot());
  test('Snapshot has user _id', () => expect(renderer.create(<NavbarUserAvatar {...{ ...defaultProps, user: { _id: 'id' } }} />).toJSON()).toMatchSnapshot());
});
