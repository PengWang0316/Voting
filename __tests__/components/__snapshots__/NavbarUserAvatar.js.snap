// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarUserAvatar Snapshot has user _id 1`] = `
<View
  style={
    Object {
      "marginRight": 15,
    }
  }
>
  Avatar
</View>
`;

exports[`NavbarUserAvatar Snapshot not user _id 1`] = `
<View
  style={
    Object {
      "marginRight": 15,
    }
  }
>
  Icon
</View>
`;
